src/lib/Scenes/Home/Components/Sales/index.tsx

```diff
@@ -1,114 +1,91 @@
-
-import { createFragmentContainer, graphql } from "react-relay"
-import styled from "styled-components/native"
-
+import createEnvironment from "lib/relay/createEnvironment"
 import fonts from "lib/data/fonts"
-import Switchboard from "lib/NativeModules/SwitchBoard"
-import SaleItem from "./Components/SaleItem"
-
-const Container = styled.View`
-  flex: 1;
-  padding: 10px 15px;
-`
-
-const SectionHeader = styled.View`
-  padding-top: 15px;
-  padding-bottom: 10px;
-  background-color: white;
-`
-
-const SectionTitle = styled.Text`
-  font-family: ${fonts["garamond-regular"]};
-  font-size: 30px;
-  text-align: left;
-  margin-left: 2px;
-`
+import styled from "styled-components/native"
+import { Dimensions, FlatList, SectionList, TouchableWithoutFeedback, View } from "react-native"
+import { LotsByFollowedArtists } from "./Components/LotsByFollowedArtists"
+import { QueryRenderer, QueryRendererProps, createFragmentContainer, graphql } from "react-relay"
+import { SaleList } from "./Components/SaleList"
+import { SectionHeader } from "./Components/SectionHeader"
+import { StyleSheet, TextStyle } from "react-native"
 
 interface Props {
   sales: Array<{
     live_start_at: string | null
   }>
 }
 
-class Sales extends React.Component<Props, null> {
-  handleTap({ item }) {
-    Switchboard.presentNavigationViewController(this, item.href)
-  }
+const SectionListStyles = StyleSheet.create({
+  contentContainer: {
+    justifyContent: "space-between",
+    padding: 10,
+    display: "flex",
+  },
+})
 
-  renderList(itemData) {
-    const numColumns = Dimensions.get("window").width > 700 ? 4 : 2
-    return (
-      <FlatList
-        contentContainerStyle={{ justifyContent: "space-between", padding: 5, display: "flex" }}
-        data={itemData.data}
-        numColumns={numColumns}
-        keyExtractor={(item, index) => item.__id}
-        renderItem={d => {
-          return (
-            <TouchableWithoutFeedback onPress={this.handleTap.bind(this, d)}>
-              <View style={{ marginRight: 10, marginBottom: 10 }}>
-                <SaleItem key={d.index} sale={d.item} />
-              </View>
-            </TouchableWithoutFeedback>
-          )
-        }}
-      />
-    )
+class Sales extends React.Component<Props> {
+  get data() {
+    const { sales } = this.props
+    const liveAuctions = sales.filter(a => !!a.live_start_at)
+    const timedAuctions = sales.filter(a => !a.live_start_at)
+
+    return {
+      liveAuctions,
+      timedAuctions,
+    }
   }
 
   render() {
-    const sales = this.props.sales
-    const liveAuctions = sales.filter(a => !!a.live_start_at)
-    const timedAuctions = sales.filter(a => !a.live_start_at)
     const sections = [
       {
-        data: [
-          {
-            data: liveAuctions,
-          },
-        ],
+        data: [{ data: this.data.liveAuctions }],
         title: "Current Live Auctions",
+        renderItem: itemData => <SaleList {...itemData} />,
```

Comment:
I moved each section into its own `renderItem` (and pushed titles into the component) so that the list can remain generic and able to render multiple types of content without much fuss

Reply:
This need jumped out when I tried to render an item that had its own `renderItem` fn, but with no title; when the list would render there was a little height gap at the bottom of the screen

Reply:
Yup, It makes more sense to have a separate `renderItem` for each section, the list used to be of homogenous item types before we added the last section

Reply:
Am I correct in understanding that there are now multiple query renderers for the one ‘auctions’ view in the home section? That would result in multiple requests, which is not what we want, we want a single query for _all_ the initial data of the view.